lighthouse/helpers/plate_event_callbacks.py

```diff
@@ -0,0 +1,107 @@
+"""Handle callbacks for registered event types
+
+Event callbacks are additional actions that will be triggered following the
+creation of an event. They are triggered by passing a Message to
+fire_callbacks.
+
+Callbacks are registered in the dictionary EVENT_TYPE_CALLBACKS, which should
+map an event_type to a callback function. Event types without callbacks
+registered will invoke _no_callback, which does nothing.
+
+Callbacks should return a Tuple[bool, List[str]] where the boolean indicates
+success, and the list contains any errors. Note that success indicates that
+the callback was processed correctly, not that it necessarily performed an
+action. For example the default _no_callback still returns a success,
+despite not doing anything, as this is the correct action for event types
+without callbacks.
+
+This file contains the following functions:
+
+  * fire_callbacks - fires the callbacks for the passed message
+"""
+
+import logging
+from lighthouse.messages.message import Message
+from typing import Tuple, List
+from lighthouse.constants import (
+    PLATE_EVENT_SOURCE_COMPLETED,
+    PLATE_EVENT_SOURCE_ALL_NEGATIVES,
+)
+from lighthouse.helpers.labwhere import set_locations_in_labwhere
+from flask import current_app as app
+
+logger = logging.getLogger(__name__)
+
+
+def fire_callbacks(event: Message) -> Tuple[bool, List[str]]:
+    """Fire any callbacks set up for a particular event type
+
+    Arguments:
+        event {Message} -- The event for which to fire a callback
+
+    Returns:
+        {bool} -- True if the operation completed successfully
+        {[str]} -- Any errors attempting to construct the message, otherwise an empty array.
+    """
+    event_type = event.event_type()
+    callback = EVENT_TYPE_CALLBACKS.get(event_type, _no_callback)
+    return callback(event)
```

Comment:
v minor thing - 
any reason why return type from this top level method is `Tuple[bool, List[str]]` while the 'private' methods return `Tuple[bool, List]`?

Cheers

Reply:
Nope, I should make them the same.